README.md

```diff
@@ -0,0 +1,133 @@
+## Memory Map
+Memory Map is a location-aware media archive that lets users upload photos, videos, and other files, automatically tagging them with time and GPS metadata, and visualizing them on an interactive world map.
+
+Users can browse the map, click markers, and explore media galleries tied to real-world locations — creating a digital memory atlas.
+
+## Features
+Upload media files (images, videos, documents)
+
+Automatic GPS location & timestamp tagging
+
+Interactive world map with clickable memory pins
+
+Gallery view for each map location
+
+GraphQL API backend
+
+Object storage powered by MinIO
+
+Rust backend with modern Nix dev-shell
+
+Fully browser-based frontend
+
+## Tech Stack
+Layer	Technology
+Frontend	React / Next.js
+Backend	Rust + GraphQL
+Storage	MinIO
+Database	PostgreSQL
+Dev Env	Nix + direnv
+Task Runner	Just
+
+## Project Structure
+
+memory-map/
+│
+├── .direnv/          # Direnv environment cache
+├── backend/         # Rust GraphQL backend
+├── data/            # Database & storage volumes
+├── devenv/          # Nix development environment
+├── frontend/        # React / Next.js frontend
+├── shared/          # Shared utilities & types
+├── .env.example     # Environment configuration template
+├── Justfile         # Development commands
+├── Cargo.toml       # Rust workspace configuration
+├── Cargo.lock       # Rust dependency lock file
+└── README.md        # Project documentation
```

Comment:
1. Please enclose this in [triple back-ticks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) to display it in a code-block and preserve your formatting.
2. Replace "Justfile" with "justfile".